File: L/cpp/code.cpp
#include <iostream>
#include <vector>

using namespace std;

struct Segment {
    int left;
    int right;

    Segment(int left, int right) : left(left), right(right) {
    }
};

vector<Segment> readSegments() {
    int n;
    cin >> n;
    vector<Segment> segments ;
    for (int i = 0; i < n; i++) {
        int left, right;
        cin >> left >> right;
        segments.push_back(Segment(left, right));
    }
    return segments;
}
    
vector<Segment> getIntersection(const vector<Segment>& firstSequence, const vector<Segment>& secondSequence) {
    // your code goes here
    return {};
}


void outputAnswer(const vector<Segment>& intersection) {
    for (const Segment& segment : intersection) {
        cout << segment.left << " " << segment.right << endl;
    }
}

int main() {
    vector<Segment> firstSequence = readSegments();
    vector<Segment> secondSequence = readSegments();

    vector<Segment> intersection = getIntersection(firstSequence, secondSequence);
    outputAnswer(intersection);
}

File: L/js/code.js
class Segment {
    constructor(left, right) {
        this.left = left;
        this.right = right;
    }
}


function getIntersection(firstSequence, secondSequence) {
    // your code goes here
    // answer for sample 1
    return [new Segment(1, 2), new Segment(4, 4), new Segment(5, 5)];
}

const _readline = require('readline');

const _reader = _readline.createInterface({
    input: process.stdin
});

const _inputLines = [];
let _curLine = 0;

_reader.on('line', line => {
    _inputLines.push(line);
});

process.stdin.on('end', solve);


function solve() {
    const firstSequence = readSegments();
    const secondSequence = readSegments(); 
    const intersection = getIntersection(firstSequence, secondSequence);
    outputAnswer(intersection);  
}

function readSegments() {
    const n = readInt();
    let result = [];
    for (let i = 0; i < n; i++) {
        let segment = readArray();
        result.push(new Segment(segment.left, segment.right));
    }
    return result;
}

function readInt() {
    const n = Number(_inputLines[_curLine]);
    _curLine++;
    return n;
}

function readArray() {
    var arr = _inputLines[_curLine].trim(" ").split(" ").map(num => Number(num));
    _curLine++;
    return arr;
}

function outputAnswer(intersection) {
    for (let segment of intersection) {
        process.stdout.write(`${segment.left} ${segment.right}\n`);
    }
}